main.go

```diff
@@ -1,19 +1,43 @@
 package main
 
 import (
+	"fmt"
 	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/helmfile/helmfile/cmd"
+	"github.com/helmfile/helmfile/pkg/app"
 	"github.com/helmfile/helmfile/pkg/config"
 	"github.com/helmfile/helmfile/pkg/errors"
 )
 
 func main() {
+	var sig os.Signal
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig = <-sigs
+	}()
```

Comment:
@yxxhero Hey! Thanks for the fix. However, I think the better strategy would be to make helm invocations `context.Context`-aware and do cancel the context at this line, after receiving a signal, so that the termination request is propagated down to all the helm processes.

Reply:
@mumoshu  yes. `context.Context` is better, but it needs to be more intrusive to the overall logic. It needs to be modified after a more detailed analysis of the logic. The current repair can be used in a transitional way, WDYT?

Reply:
Good suggestion! Yeah maybe... Just confirming but how does this actually propagate signals from Helmfile to Helm processes? Or, is this meant to let Helmfile just wait until Helm processes exit?

Reply:
By default, `cmd.Exec` or `cmd.Command` can receive signals, but according to the old logic, the helmfile will directly by os.Exit does not wait for the return of the exit after the HELM receives the signal. Currently, this part is only optimized. Wait for the exit of the HELM receiving signal, and then OS Exit (130), while waiting for all clean logic to complete.

Reply:
Makes sense! I think this can be mergeable once https://github.com/helmfile/helmfile/pull/418#discussion_r990594856 is confirmed and tested.